src/Core/Module/ModuleRepository.php

```diff
@@ -282,7 +282,11 @@ private function getModulesFromHook()
         if ($this->modulesFromHook === null) {
             $modulesFromHook = $this->hookManager->exec('actionListModules', [], null, true);
             $modulesFromHook = array_values($modulesFromHook ?? []);
-            $this->modulesFromHook = empty(reset($modulesFromHook)) ? [] : array_merge(...$modulesFromHook);
+
+            $this->modulesFromHook = [];
+            foreach ($modulesFromHook as $modulesFromSingleHook) {
+                $this->modulesFromHook += $modulesFromSingleHook;
+            }
```

Comment:
I think you'll run into an issue if you try to add an empty string to an array. I guess you can filter the array:
```suggestion
            $this->modulesFromHook = array_merge(...array_filter($modulesFromHook, function ($item) { return is_array($item); }));
```